src/test/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperatorTest.scala

```diff
@@ -3,210 +3,64 @@ package tech.cryptonomic.conseil.tezos
-      scala.io.Source.fromFile(s"src/test/resources/tezos_blocks/$hash.account.$accountID.json").mkString
-  }
-
-  object MockTezosNodeWithErrors extends TezosRPCInterface {
+import scala.concurrent.{Await, ExecutionContext, Future}
 
-    def runGetQuery(network: String, command: String, payload: Option[String] = None): Try[String] = Try {
-      command match {
-        case "blocks/BKiRLq7c2QVr6X428RRvp6JLTJEnWPE4bc4cAQHoo9GuZz9GH38" =>
-          throw new Exception("A block request failed due to an alien invasion.")
-        case _ => MockTezosNode.runGetQuery(network, command).get
-      }
-    }
+class TezosNodeOperatorTest extends FlatSpec with MockFactory with Matchers with LazyLogging {
 
-  }
+  val tezosRPCInterface = stub[TezosRPCInterface]
+  val config = BatchFetchConfiguration(1, 1)
 
-  val keyStore: KeyStore = KeyStore(
-    publicKey = "edpkv3azzeq9vL869TujYhdQY5FKiQH4CGwJEzqG7m6PoX7VEpdPc9",
-    privateKey = "edskS5owtVaAtWifnCNo8tUpAw2535AXEDY4RXBRV1NHbQ58RDdpaWz2KyrvFXE4SuCTbHU8exUecW33GRqkAfLeNLBS5sPyoi",
-    publicKeyHash = "tz1hcXqtiMYFhvuirD4guE7ts4yDuCAmtD95"
-  )
+  implicit val executionContext = ExecutionContext.global
 
   "getBlock" should "should correctly fetch the genesis block" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNode)
-    val block: Try[TezosTypes.Block] = nodeOp.getBlock("zeronet", "BLockGenesisGenesisGenesisGenesisGenesis385e5hNnQTe")
-    block.get.metadata.hash should be ("BLockGenesisGenesisGenesisGenesisGenesis385e5hNnQTe")
-  }
-
-  //skip block at level 1 because zeronet doesn't give proper response, also, hint because not the head
-  "getBlocks" should "fetch the correct number of blocks" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNode)
-    val blocks = nodeOp.getBlocks("zeronet", 0, 2, None, followFork = false)
-    blocks.get.length should be (3)
-  }
-
-  "getBlocks" should "handle a failed RPC request" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNodeWithErrors)
-    val blocks = nodeOp.getBlocks("zeronet", 0, 5, None, followFork = false)
-    blocks.isFailure should be (true)
-  }
-
-  "getBlocks" should "work correctly with a hint whose level is too low" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNode)
-    val blocks = nodeOp.getBlocks("zeronet", 1, 2, Some("BLockGenesisGenesisGenesisGenesisGenesis385e5hNnQTe"), followFork = false)
-    blocks.get.length should be (0)
-  }
-
-  "getBlocks" should "handle an invalid block payload" in {
-    val nodeOp: TezosNodeOperator = new TezosNodeOperator(MockTezosNode)
-    val blocks = nodeOp.getBlocks("zeronet", 0, 2, Some("fakeblock"), followFork = false)
-    blocks.isFailure should be (true)
-  }
+    //given
+    (tezosRPCInterface.runAsyncGetQuery _).when("zeronet", "blocks/BLockGenesisGenesisGenesisGenesisGenesis385e5hNnQTe~").returns(Future.successful(TezosResponseBuilder.blockResponse))
```

Comment:
Very subjective: I personally try to avoid using externally defined classes to provide canned values for a specific test, or would try to make it very clear that the response-builder is strictly connected to this specific spec.

Rationale: I've seen how creating such test utilities can entice all devs in the team to build up on the same utility class, which will grow in time to a monster file that will be required to somehow provide a consistent view for all possible sorts of tests, becoming a scourge to maintain...
That's why I try to keep such things defined near the spec at hand, or, when needed, create a more generic utility that could be parametrized progressively to suit each case with explicit configuration.
The pain point is returning highly tailored test-specific data in a seemingly innocuous helper class.

I'd appreciate feedback on this as a chance to improve myself in my testing design skills.